File: JAVA/198. House Robber.java
class Solution {
    public int rob(int[] count) {
        if(count == null|| count.length==0)
        return 0;
        int i;
        for(i=1;i<count.length;i++)
        {
            if(i==1)
            {
                count[i]=Math.max(count[0],count[1]);
            }
            else
            {
                count[i]=Math.max(count[i-1],count[i]+count[i-2]);
            }
        }
        return count[count.length-1];
    }
}

File: C++/198. House Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int f0 = 0, f1 = 0;
        for (auto& x : nums) {
            int f2 = max(f0+x, f1); 
            f0 = f1; 
            f1 = f2;
        }
        return f1; 
    }
};
